R/CreatePackageVignette.R

```diff
@@ -0,0 +1,101 @@
+#' @importFrom R6 is.R6Class
+#' @importFrom glue glue
+#' @export
+CreatePackageVignette <- function(pkg_name
+                                  , pkg_path = NULL
+                                  , pkg_reporters = list(
+                                      DependencyReporter$new()
+                                      , FunctionReporter$new()
+                                  )
+                                  , vignette_path = file.path("vignettes"
+                                                              , "pkgnet_report.Rmd")
+                                 ) {
+
+    # Capture pkg_reporters expression for later injection into Rmd
+    pkg_reporters_expr <- rlang::enexpr(pkg_reporters)
+
+    ## pkg_name input checks ##
+    assertthat::assert_that(
+        assertthat::is.string(pkg_name)
+        , pkg_name != ""
+    )
+
+    ## pkg_path input checks ##
+    assertthat::assert_that(
+        is.null(pkg_path) || assertthat::is.readable(pkg_path)
+    )
+
+    ## vignette_path input checks ##
+    assertthat::assert_that(
+        assertthat::is.string(vignette_path)
+        , vignette_path != ""
+        , identical(tolower(tools::file_ext(vignette_path)), "rmd")
+    )
+    # Confirm directory exists
+    if (!assertthat::is.dir(dirname(vignette_path))) {
+        log_fatal(sprintf("directory %s does not exist, please create first"))
+    }
+
+    ## pkg_reporter input checks ##
+    assertthat::assert_that(
+        is.list(pkg_reporters)
+    )
+    # Check if generators were passed in by accident
+    if (any(vapply(pkg_reporters, FUN = R6::is.R6Class, FUN.VALUE = logical(1)))) {
+        log_fatal("At least one of pkg_reporters is an R6 class generator. This
+                  function expects initialized reporter objects.")
+    }
+    # Confirm that all reporters are actually valid initialized reporters
+    assertthat::assert_that(
+        all(vapply(pkg_reporters
+                   , FUN = .is.PackageReporter
+                   , FUN.VALUE = logical(1)
+            ))
+    )
+
+    # If pkg_path supplied, add quotes, otherwise, set to NULL as a string
+    if (!is.null(pkg_path)) {
+        pkg_path <- paste0("\"", pkg_path, "\"")
+    } else {
+        pkg_path <- "NULL"
+    }
+
+    # Read pkgnet vignette template
+    templatePath <- system.file(file.path("package_report"
+                                          , "package_vignette_template.Rmd")
+                                , package = "pkgnet")
+
+    # Inject code into the template
+    vignette_rmd <- glue::glue(
+        paste(readLines(templatePath), collapse = "\n")
+        , pkg_name = pkg_name
+        , pkg_path = pkg_path
+        , pkg_reporters = deparse(pkg_reporters_expr)
+        , .open = "{{"
+        , .close = "}}"
+    )
+
+    # Write vignette Rmd to file
+    rmd_conn <- file(description = vignette_path, open = 'w')
+    on.exit(close(rmd_conn))
+    writeLines(vignette_rmd, con = rmd_conn)
+
+    return(invisible(NULL))
```

Comment:
Consider a success message to terminal.

Reply:
@jayqi any thoughts on this?

Reply:
we talked IRL but just documenting here...should log out "you can view this vignette at /fully/qualified/path/to/whatever"

Reply:
✅